pkg/reaper/deployment.go

```diff
@@ -169,121 +158,246 @@ func NewDeployment(reaper *api.Reaper, dc *cassdcapi.CassandraDatacenter, keysto
 		},
 	}
-	volumes := []corev1.Volume{
-		{
-			Name: "conf",
+
+	if reaper.Spec.HttpManagement.Enabled && reaper.Spec.HttpManagement.Keystores != nil {
+		volumes = append(volumes, corev1.Volume{
+			Name: "management-api-keystore",
 			VolumeSource: corev1.VolumeSource{
-				EmptyDir: &corev1.EmptyDirVolumeSource{},
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: reaper.Spec.HttpManagement.Keystores.Name,
+				},
+			},
+		})
+
+		volumeMounts = append(volumeMounts, corev1.VolumeMount{
+			Name:      "management-api-keystore",
+			MountPath: "/etc/encryption/mgmt",
+		})
+	}
+
+	if reaper.Spec.StorageType == api.StorageTypeLocal {
+		volumes = append(volumes, corev1.Volume{
+			Name: "reaper-data",
+			VolumeSource: corev1.VolumeSource{
+				PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+					ClaimName: fmt.Sprintf("reaper-data-%s", reaper.Name),
+				},
+			},
+		})
+		volumeMounts = append(volumeMounts, corev1.VolumeMount{
+			Name:      "reaper-data",
+			MountPath: "/var/lib/cassandra-reaper/storage",
+		})
+	}
+
+	return volumes, volumeMounts
+}
+
+func makeSelector(reaper *api.Reaper) *metav1.LabelSelector {
+	return &metav1.LabelSelector{
+		MatchExpressions: []metav1.LabelSelectorRequirement{
+			// Note: managed-by shouldn't be used here, but we're keeping it for backwards compatibility, since changing
+			// a deployment's selector is a breaking change.
+			{
+				Key:      v1alpha1.ManagedByLabel,
+				Operator: metav1.LabelSelectorOpIn,
+				Values:   []string{v1alpha1.NameLabelValue},
+			},
+			{
+				Key:      api.ReaperLabel,
+				Operator: metav1.LabelSelectorOpIn,
+				Values:   []string{reaper.Name},
 			},
 		},
 	}
+}
+
+func makeObjectMeta(reaper *api.Reaper) metav1.ObjectMeta {
+	return metav1.ObjectMeta{
+		Namespace:   reaper.Namespace,
+		Name:        reaper.Name,
+		Labels:      createServiceAndDeploymentLabels(reaper),
+		Annotations: map[string]string{},
+	}
+}
+
+func computePodMeta(reaper *api.Reaper) metav1.ObjectMeta {
+	podMeta := getPodMeta(reaper)
+	return metav1.ObjectMeta{
+		Labels:      podMeta.Labels,
+		Annotations: podMeta.Annotations,
+	}
+}
+
+func configureClientEncryption(reaper *api.Reaper, envVars *[]corev1.EnvVar, volumes *[]corev1.Volume, volumeMounts *[]corev1.VolumeMount, keystorePassword *string, truststorePassword *string) {
 	// if client encryption is turned on, we need to mount the keystore and truststore volumes
+	// by client we mean a C* client, so this is only relevant if we are making a Deployment which uses C* as storage backend
 	if reaper.Spec.ClientEncryptionStores != nil && keystorePassword != nil && truststorePassword != nil {
 		keystoreVolume, truststoreVolume := cassandra.EncryptionVolumes(encryption.StoreTypeClient, *reaper.Spec.ClientEncryptionStores)
-		volumes = append(volumes, *keystoreVolume)
-		volumeMounts = append(volumeMounts, corev1.VolumeMount{
+		*volumes = append(*volumes, *keystoreVolume)
```

Comment:
issue: Okay, lets stop here. This is not looking like the correct way to use pointers, since you keep deferring for every operation. And I don't think you really need pointers here, the slices themselves are always pointers and also copied by value.

So, the data inside the slice is still the same, just the slice information (length) could be from wrong one. 

Just do normal volumes slice, normal volumeMounts slices and normal envVars. Set those new values somewhere to the correct place (if even necessary). 

Even right now *enVars = append(*envVars) has already new version of envVars due to how append works.